libs/ng-mocks/src/lib/resolve/collect-declarations.ts

```diff
@@ -316,6 +323,42 @@ const parseNgDef = (
   }
 };
 
+/**
+ * Note: This does not seem to work in every environment (e.g. the tests)
+ *       and is therefore a supplementary support for signals.
+ */
+const parseReflectComponentType = (def: any, declaration: Declaration): void => {
+  if (typeof def === 'function') {
+    try {
+      const mirror = reflectComponentType(def);
```

Comment:
If you have suggestions around automated testing and/or manual QA, I'm happy to follow-up. Or even alternative solutions to explore all-together. It's open-source so I want to take as much of the burden off of you as possible :pray: